app/controllers/deployments_controller.rb

```diff
@@ -48,10 +54,32 @@ def create
 private
 
   def application_by_repo
-    if (existing_app = Application.find_by(repo: repo_path))
-      existing_app
+    existing_apps = Application.where(repo: repo_path)
+    if existing_apps.present?
+      if existing_apps.length == 1
+        existing_apps[0]
+      else
+        flash[:alert] = format("Found multiple applications using repo: %<repo_path>s while using application_by_repo",
+                               repo_path: repo_path)
+        render :new
+      end
+    else
+      Application.create!(name: app_name(repo_path), repo: repo_path, domain: domain)
+    end
+  end
+
+  def application_by_name
+    existing_apps = Application.where(repo: repo_path, name: params[:application_name])
+
+    case existing_apps.length
+    when 1
+      existing_apps[0]
+    when 0
+      Application.create!(name: app_name(params[:application_name]), repo: repo_path, domain: domain)
     else
-      Application.create!(name: app_name, repo: repo_path, domain: domain)
+      flash[:alert] = format("Found multiple applications using repo: %<repo_path>s and application name: %<application_name>s",
+                             repo_path: repo_path, application_name: params[:application_name])
+      render :new
```

Comment:
Given the application name is unique, I don't think this can happen right?

Reply:
I will do this